File: my-folder/0014-longest-common-prefix/solution.cpp
class Solution {
public:

    string longestCommonPrefix(vector<string>& strs) 
    {
        string ans="";
        for (int i=0; i<strs[0].size();i++)
        {
            char f = strs[0][i];
            bool match = true;

            for (int j=1; j<strs.size(); j++)
            {   
                if (strs[j][i] != f)
                {
                    match = false;
                }
            }
            if (match)
            {
                ans+=f;
            }
            else
            {
                return ans;
            }
        }
        return ans;
    }
};

File: my-folder/0014-longest-common-prefix/solution.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        
        n = len(strs)


        ans = []
        first = strs[0]

        for index,char in enumerate(first):
            success = True
            for i in range(1, n):
                if len(strs[i]) > index and strs[i][index] == char:
                    continue
                
                else:
                    success = False
                    break
            if success:
                ans.append(char)
            else:
                break
        
        return "".join(ans) if len(ans) > 0 else ""
            


